Animations/src/platform/PlatformLinux.cpp

```diff
@@ -0,0 +1,302 @@
+#ifdef __linux
+#include "platform/Platform.h"
+#include "core.h"
+
+#include <filesystem>
+
+#include <sys/wait.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <pwd.h>
+
+#include <openssl/md5.h>
+
+#ifdef min
+#undef min
+#endif
+
+/* Make a directory; already existing dir okay */
+static int maybe_mkdir(const char *path, mode_t mode)
+{
+  struct stat st;
+  errno = 0;
+
+  /* Try to make the directory */
+  if (mkdir(path, mode) == 0)
+    return 0;
+
+  /* If it fails for any reason but EEXIST, fail */
+  if (errno != EEXIST)
+    return -1;
+
+  /* Check if the existing path is a directory */
+  if (stat(path, &st) != 0)
+    return -1;
+
+  /* If not, fail with ENOTDIR */
+  if (!S_ISDIR(st.st_mode))
+  {
+    errno = ENOTDIR;
+    return -1;
+  }
+
+  errno = 0;
+  return 0;
+}
+
+static int mkdir_p(const char *path, mode_t mode)
+{
+  /* Adapted from http://stackoverflow.com/a/2336245/119527 */
+  char *_path = NULL;
+  char *p;
+  int result = -1;
+
+  errno = 0;
+
+  /* Copy string so it's mutable */
+  _path = strdup(path);
+  if (_path == NULL)
+    goto out;
+
+  /* Iterate the string */
+  for (p = _path + 1; *p; p++)
+  {
+    if (*p == '/')
+    {
+      /* Temporarily truncate */
+      *p = '\0';
+
+      if (maybe_mkdir(_path, mode) != 0)
+        goto out;
+
+      *p = '/';
+    }
+  }
+
+  if (maybe_mkdir(_path, mode) != 0)
+    goto out;
+
+  result = 0;
+
+out:
+  free(_path);
+  return result;
+}
+
+namespace MathAnim
+{
+  namespace Platform
+  {
+    // --------------- Internal Functions ---------------
+    // static void wideToChar(const WCHAR *wide, char *buffer, size_t bufferLength);
+    // static bool stringsEqualIgnoreCase(const char *str1, const char *str2);
+    static std::vector<std::string> availableFonts = {
+        "JetBrains Mono"};
+    static bool availableFontsCached = false;
+    static std::string homeDirectory = std::string();
+
+    const std::vector<std::string> &getAvailableFonts()
+    {
+      if (!availableFontsCached)
+      {
+        // TODO: Placeholder cuz it's 01:39
+        availableFontsCached = true;
+      }
+
+      return availableFonts;
+    }
+
+    // Adapted from https://stackoverflow.com/questions/2467429/c-check-installed-programms
+    bool isProgramInstalled(const char *displayName)
+    {
+      // TODO: Placeholder, but also this function seems unused?
+      return false;
+    }
+
+    bool getProgramInstallDir(const char *programDisplayName, char *buffer, size_t bufferLength)
+    {
+      // TODO: I'm used to Rust or managed languages and string handling in C/++ scares me
+
+      std::string const path = std::getenv("PATH");
```

Comment:
This is like a face palm haha. I don't know why I didn't think of this on Windows, but now I want to go modify the code and add in support for this :D

Reply:
Certainly better than prodding the registry
Especially if you have a portable latex install (which I'm just asserting exists)